applications/Chat/evaluate/metrics.py

```diff
@@ -72,19 +119,35 @@ def distinct_score(preds: list) -> dict:
     cumulative_distinct = []
 
     for pred in preds:
-        pred_seg_list = list(' '.join(jieba.cut(pred)))
-        count_segs = len(pred_seg_list)
-        unique_segs = set(pred_seg_list)
-        count_unique_chars = len(unique_segs)
+        if language == "cn":
+            pred_seg_list = ' '.join(jieba.cut(pred)).split()
+            count_segs = len(pred_seg_list)
+            unique_segs = set(pred_seg_list)
+            count_unique_chars = len(unique_segs)
+
+            cumulative_distinct.append(count_unique_chars / count_segs)
+        elif language == "en":
+            # calculate distinct 1-gram, 2-gram, 3-gram
+            unique_ngram = [set() for _ in range(0, 3)]
+            all_ngram_count = [0 for _ in range(0, 3)]
+
+            split_pred = preprocessing_text(pred).split()
+            for n in range(0, 3):
+                for i in range(0, len(split_pred) - n):
+                    ngram = ' '.join(split_pred[i:i + n + 1])
+                    unique_ngram[n].add(ngram)
+                    all_ngram_count[n] += 1
+
+            avg_distinct = [len(a) / (b + 1e-6) for a, b in zip(unique_ngram, all_ngram_count)]
```

Comment:
Please add one comment to explain this, any ref why do we use 1e-6 ?

Reply:
Sometimes the answer may contain only one word. For 2-gram and 3-gram, the gram count(denominator) may be zero.